### ISSUE TITLE
Recording does not stop

### ISSUE THREAD

**Issue by parvuselephantus** [bryanjenningz/record-audio#16]
I'm sorry to say, that recording doesn't stop under Mozilla Firefox. I tried multiple times - it worked just a few. Usually it just keeps recording until page refresh.

**Comment by bryanjenningz** [comment]
Thanks for opening this issue up. I'll try to help you fix this. I tried running the code in Firefox just now and it works for me so I'm not able to reproduce this problem.

Here are the steps I took to get it working.

First, I ran these commands in my terminal to Git clone the repo and open up the index.html file in Firefox.
```bash
git clone https://github.com/bryanjenningz/record-audio.git
cd ./record-audio
open ./index.html
```

Then I clicked on the "Start recording..." button on the index.html in Firefox.
![recording-test-page](https://user-images.githubusercontent.com/7637655/236022811-d6e7b82f-ba05-4f32-b27a-9420e6443f30.png)

Then I got a prompt from Firefox asking for permission to record audio and I clicked "Allow".
![page-asks-for-audio-permission](https://user-images.githubusercontent.com/7637655/236022999-64842c1f-9814-4053-beef-267d8fe661ef.png)

Then it should record for about 3 seconds so I spoke into my microphone and then after 3 seconds Firefox should play back your recording.

Make sure your microphone and speakers are turned on so you can hear the recording.

If there are still problems, try opening up the Firefox developer tools and see if there are any errors in the console.

Let me know if you are still having any difficulties and I'll try to help.

**Comment by parvuselephantus** [comment]
Thank you for your quick reply. I tried few tutorials and all of them worked same for me. I'm afraid it is also somehow related to headphones I'm using - these are bose 700 bluetooth headphones, which have special mode for call and regular music. I suspect, that when turning microphone on they get stuck. Even though I get all the logs, that recording is stopping/stopped in browsers console, I still hear heapphones are in recording mode and browser is still having red microphone icon (which means it records). The speach recording object is in "inactive" status. There is a way to get a link to the recording - fun fact: it works and indeed I hear 3 second long recording there... but browser thinks I'm still recording (till page refresh).
To sum up - I'm sorry for bodering you, it indeed seems it's my configuration I don't know how to control.

**Comment by bryanjenningz** [comment]
Thanks for pointing this problem out.

I understand what you meant now when you said the browser thinks you're still recording - the browser displays a microphone icon on the browser tab that indicates that the browser is still using the microphone even though you aren't recording anymore. I'm able to reproduce this problem on my computer.

![Screenshot_20230504_163724](https://user-images.githubusercontent.com/7637655/236352260-636b83ff-72ff-42be-a7f3-cea0a5684a3e.png)

I think the reason why this happens is because there are still tracks in the media stream that have not been stopped so Firefox thinks that we are still recording. To fix this, we can add one line of code to stop all the media stream's tracks when the user stops recording:
```js
stream.getTracks().forEach((track) => track.stop());
```

Adding this line of code seems to fix the problem for me. I made a pull request which adds this change: https://github.com/bryanjenningz/record-audio/pull/17. Feel free to apply this change to your code and see if it fixes the problem for you too. If it fixes the problem then feel free to approve the pull request and then I'll merge the pull request.

**Comment by parvuselephantus** [comment]
It works!

This is amazing - I had very this issue when following a few tutorials (like nobody cared about cleanup). Tried searching for these opened lines, but unfortunately API is new to me and I failed. Thank you for your help!
